chapters/memory-layout/memory-operations/reading/README.md

```diff
@@ -0,0 +1,329 @@
+> Declaring a pointer does not mean allocating a memory area to store data. A pointer is also a data type, whose value is a number representing a memory address.
+>
+> ```c
+>   int *p = 0xCAFEBABE; /* Declaring a pointer */
+>   int x = *p; /* The value at the address stored in p */
+> ```
+
+In C, a pointer can represent:
+
+- The address of data of a certain type
+- The address of a memory area
+- The address of a function
+- The address of an area with unknown content (void pointer)
+
+> **TIP:** The size of a pointer depends on the architecture and operating system on which the program was compiled. The size of a pointer is determined by `sizeof(void*)` and is not necessarily equal to the size of an `int`.
+
+## **Pointer Operations and Pointer Arithmetic**
+
+Arithmetic operations on pointers are slightly different from those on integer data types. The only valid operations are **incrementing** or **decrementing** a pointer, **adding** or **subtracting** an integer from a pointer, and subtracting two pointers of the **same type**. The behavior of these operations is influenced by the data type to which the pointers refer.
+
+When incrementing a pointer related to a data type `T`, the address is not increased by 1 but by the value `sizeof(T)`, which ensures addressing the next object of the same type. Similarly, adding an integer `n` to a pointer `p` (thus the operation `p + n`) actually represents `p + n * sizeof(*p)`. For example:
+
+```c
+char *char_ptr = 1000;
+short *short_ptr = 2000;
+int *int_ptr = 3000;
+
+++char_ptr; /* char_ptr will point to address 1001 */
+++short_ptr; /* short_ptr points to address 2002 */
+++int_ptr; /* int_ptr points to address 3004 */
+```
+
+![A diagram which visualizes arithmetic operations on pointers](../media/arit.svg)
+
+Subtracting two pointers is possible only if both have the same type. The result of the subtraction is obtained by calculating the difference between the memory addresses they point to. For example, calculating the length of a string:
+
+```c
+char *s = "Learn IOCLA, you must!";
+char *p = s;
+for (; *p; ++p); /* Iterating character by character until '\0' */
+
+printf("%ld", p - s); /* It will display 22. */
+```
+
+### **Interpreting Data in Memory**
+
+On most modern computers, the smallest unit of data that can be addressed is the `byte` (8 bits), meaning that we can view data in memory as a sequence of bytes, each with its own address. As mentioned in the [previous lab](https://ocw.cs.pub.ro/courses/pclp2/laboratoare/laborator-01), when we want to store information represented by multiple bytes, we need to consider the order imposed by the system architecture, called [endianness](https://en.wikipedia.org/wiki/Endianness). Below is the mechanism for extracting data from memory on a **little-endian** architecture:
```

Comment:
One thing I wasn't sure how to approach is the link to the previous lab in this paragraph, since the previous lab hasn't been published yet, meaning this PR depends on #1 being merged first. For now it just points to the old lab 1 deployment.

Reply:
Use the path to the previous lab's Markdown file. Now I've just merged the previous lab. Rebase this PR on top of the updated `main` branch and the link should work.